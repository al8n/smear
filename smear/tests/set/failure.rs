
#[smear::hashset(
  value(String)
)]
enum Foo {

}


#[smear::btreeset(
  value(String)
)]
struct Bar(f64, u32);


#[smear::hashset(
  value("String"),
  attributes(derive(Clone))
)]
struct Baz {
  f1: u64,
  f2: u32,
}

#[smear::btreeset(
  value(String)
)]
struct Bazz(u64);

fn main() {}