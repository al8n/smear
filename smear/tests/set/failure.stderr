error: Unsupported shape `enum`. Expected struct with no fields.
 --> tests/set/failure.rs:2:1
  |
2 | / #[smear::hashset(
3 | |   value(String)
4 | | )]
  | |__^
  |
  = note: this error originates in the attribute macro `smear::hashset` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unsupported shape `unnamed fields`. Expected no fields.
  --> tests/set/failure.rs:10:1
   |
10 | / #[smear::btreeset(
11 | |   value(String)
12 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `smear::btreeset` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unsupported shape `named fields`. Expected no fields.
  --> tests/set/failure.rs:16:1
   |
16 | / #[smear::hashset(
17 | |   value("String"),
18 | |   attributes(derive(Clone))
19 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `smear::hashset` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unsupported shape `one unnamed field`. Expected no fields.
  --> tests/set/failure.rs:25:1
   |
25 | / #[smear::btreeset(
26 | |   value(String)
27 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `smear::btreeset` (in Nightly builds, run with -Z macro-backtrace for more info)
